http/httpRequest.h

```diff
@@ -42,18 +42,25 @@ class Request {
     return this->headers_;
   }
 
+	void SetURI(const std::string& uri);
```

Comment:
but I think we use instances of regular httpRequest class...so those can't be purely virtual or we have to implement another subclass of httpRequest